mtg_card_identifier/CardScannerFile.swift

```diff
@@ -0,0 +1,125 @@
+//
+//  CardScannerFile.swift
+//  mtg_card_identifier
+//
+//  Created by mac on 8/16/23.
+//
+
+import SwiftUI
+import UIKit
+import VisionKit
+
+struct CardScanner: UIViewControllerRepresentable {
+    
+    //    @Binding var startScanning: Bool
+    @Binding var scanedCard: [String]
+    
+    func makeCoordinator() -> Coordinator {
+        Coordinator(self)
+    }
+    
+    func makeUIViewController(context: Context) -> DataScannerViewController {
+        
+        let vc = DataScannerViewController(
+            recognizedDataTypes: [.text()],
+            qualityLevel: .balanced,
+            recognizesMultipleItems: true,
+            isHighFrameRateTrackingEnabled: true
+            //            isGuidanceEnabled: true,
+//            Don't know if we want that to work but it's basically done
+//            isHighlightingEnabled: true
+        )
+        let partialTransparentView = PartialTransparentView(cutout: CGRect(x: 31, y: 75, width: 350, height: 470))
+        partialTransparentView.translatesAutoresizingMaskIntoConstraints = true
+        partialTransparentView.autoresizingMask = [.flexibleWidth, .flexibleHeight]
+        partialTransparentView.frame = vc.overlayContainerView.bounds
+        
+        vc.view.addSubview(partialTransparentView)
+        vc.regionOfInterest = CGRect(x: 31, y: 75, width: 350, height: 430)
+        vc.delegate = context.coordinator
+        
+        return vc
+    }
+    
+    func updateUIViewController(_ uiViewController: DataScannerViewController, context: Context) {
+        do {
+            try uiViewController.startScanning()
```

Comment:
We probably want this to be up in the makeUIViewController function. If that's a problem, then we'd want some way to see if it is scanning already or not and only `startScanning` if it is not already scanning.

Reply:
I added the startScanning to the vc of makeUIViewController and that seems to work but the delegate UIViewControllerRepresentable needs the updateUIViewController so i just didn't put any code in it

Reply:
Yeah that's fine to have it empty.